src/Custom/Microsoft.Extensions.AI.OpenAIAdapter.cs

```diff
@@ -0,0 +1,1820 @@
+    {
+        foreach (ChatMessage input in inputs)
+        {
+            if (input.Role == ChatRole.System ||
+                input.Role == _chatRoleDeveloper)
+            {
+                string text = input.Text;
+                if (!string.IsNullOrWhiteSpace(text))
+                {
+                    yield return input.Role == ChatRole.System ?
+                        ResponseItem.CreateSystemMessageItem(text) :
+                        ResponseItem.CreateDeveloperMessageItem(text);
+                }
+
+                continue;
+            }
+
+            if (input.Role == ChatRole.User)
+            {
+                yield return ResponseItem.CreateUserMessageItem(ToOpenAIResponsesContent(input.Contents));
+                continue;
+            }
+
+            if (input.Role == ChatRole.Tool)
+            {
+                foreach (AIContent item in input.Contents)
+                {
+                    switch (item)
+                    {
+                        case FunctionResultContent resultContent:
+                            string? result = resultContent.Result as string;
+                            if (result is null && resultContent.Result is not null)
+                            {
+                                try
+                                {
+                                    result = JsonSerializer.Serialize(resultContent.Result, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(object)));
+                                }
+                                catch (NotSupportedException)
+                                {
+                                    // If the type can't be serialized, skip it.
+                                }
+                            }
+
+                            yield return ResponseItem.CreateFunctionCallOutputItem(resultContent.CallId, result ?? string.Empty);
+                            break;
+                    }
+                }
+
+                continue;
+            }
+
+            if (input.Role == ChatRole.Assistant)
+            {
+                foreach (AIContent item in input.Contents)
+                {
+                    switch (item)
+                    {
+                        case TextContent textContent:
+                            yield return ResponseItem.CreateAssistantMessageItem(textContent.Text);
+                            break;
+
+                        case FunctionCallContent callContent:
+                            yield return ResponseItem.CreateFunctionCallItem(
+                                callContent.CallId,
+                                callContent.Name,
+                                BinaryData.FromBytes(JsonSerializer.SerializeToUtf8Bytes(
+                                    callContent.Arguments,
+                                    AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(IDictionary<string, object?>)))));
+                            break;
+                    }
```

Comment:
It occurs to me that although we're creating ErrorContent when processing responses output, we're not here translating that ErrorContent back into something in the ResponseItems. I don't know if there's a way to do so, but if there is, we probably should.